core/trino-main/src/main/java/io/trino/connector/CatalogConnector.java

```diff
@@ -89,6 +105,8 @@ public ConnectorServices getMaterializedConnector(CatalogHandleType type)
 
     public void shutdown()
     {
+        long bytes = catalogConnector.getConnector().getInitialMemoryRequirement();
```

Comment:
This is still problematic. The shutdown method should free up *exactly* the number of bytes that were reserved in the constructor above. The current implementation is sensitive to connectors returning different values across different invocations of `getInitialMemoryRequirements()`.

Reply:
@martint you are suggesting to keep the value at the creation of the Catalog and use it when shutdown?

Reply:
Yes, exactly